MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -0,0 +1,1080 @@
+            scenario = generateScenario(campaign, contract, track, forceID, coords);
+            
+            // we may generate a facility scenario randomly - if so, do the facility-related stuff
+            // and add a new facility to the track
+            if(scenario.getBackingScenario().getTemplate().isFacilityScenario()) {
+                StratconFacility facility = scenario.getBackingScenario().getTemplate().isHostileFacility() ?
+                        StratconFacilityFactory.getRandomHostileFacility() : StratconFacilityFactory.getRandomAlliedFacility();
+                facility.setVisible(true);
+                track.addFacility(coords, facility);
+                setupFacilityScenario(scenario, track, coords, facility);
+            }
+        }
+        
+        return scenario;
+    }
+    
+    /**
+     * carries out tasks relevant to facility scenarios
+     */
+    private static void setupFacilityScenario(StratconScenario scenario, StratconTrackState track, StratconCoords coords, StratconFacility facility) {
+        // this includes:
+        // for hostile facilities
+        // - add a destroy objective (always the option to level the facility)
+        // - add a capture objective (always the option to capture the facility)
+        // - if so indicated by parameter, roll a random hostile facility objective and add it if not capture/destroy
+        // for allied facilities
+        // - add a defend objective (always the option to defend the facility)
+        // - if so indicated by parameter, roll a random allied facility objective and add it if not defend
+        AtBScenarioModifier objectiveModifier = null;
+        boolean alliedFacility = facility.getOwner() == ForceAlignment.Allied;
+        
+        if(alliedFacility) {
+            objectiveModifier = AtBScenarioModifier.getRandomAlliedFacilityModifier();
+        } else {            
+            objectiveModifier = AtBScenarioModifier.getRandomHostileFacilityModifier();
+        }
+        
+        if(objectiveModifier != null) {
+            scenario.getBackingScenario().addScenarioModifier(objectiveModifier);
+            scenario.getBackingScenario().setName(String.format("%s - %s - %s", 
+                    facility.getFacilityType(), alliedFacility ? "Allied" : "Hostile", objectiveModifier.getModifierName()));
+        }
+        
+        // add the "fixed" hostile facility modifiers after the primary ones
+        if(!alliedFacility) {
+            for(AtBScenarioModifier modifier : AtBScenarioModifier.getRequiredHostileFacilityModifiers()) {
+                if(!scenario.getBackingScenario().alreadyHasModifier(modifier)) {
+                    scenario.getBackingScenario().addScenarioModifier(modifier);
+                }
+            }
+        }
+    }
+    
+    private static void processFacilityEffects(StratconTrackState track) {
+        // todo: these are "weekly"? effects that a stratcon facility has on the campaign/track state
```

Comment:
So my IDE will highlight these :)
```suggestion
        // TODO: these are "weekly"? effects that a stratcon facility has on the campaign/track state
```